core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -17,149 +17,175 @@
  * question domain objects.
  */
 
-require('domain/state/StateObjectFactory.ts');
-
-angular.module('oppia').factory('QuestionObjectFactory', [
-  'StateObjectFactory', 'DEFAULT_LANGUAGE_CODE', 'INTERACTION_SPECS',
-  function(StateObjectFactory, DEFAULT_LANGUAGE_CODE, INTERACTION_SPECS) {
-    var Question = function(id, stateData, languageCode, version,
-        linkedSkillIds) {
-      this._id = id;
-      this._stateData = stateData;
-      this._languageCode = languageCode;
-      this._version = version;
-      this._linkedSkillIds = linkedSkillIds;
-    };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { InteractionSpecsConstants } from 'pages/interaction-specs.constants';
+import { StateObjectFactory, State } from 'domain/state/StateObjectFactory';
+
+export interface IQuestionBackendDict {
+  id: string;
+  'question_state_data': State;
+  'language_code': string;
+  version: number;
+  'linked_skill_ids': string[];
+}
+
+export class Question {
+  _id: string;
```

Comment:
Are these supposed to be private? If so, just give them a private modifier and get rid of the leading underscore.

Reply:
@brianrodri Underscores are removed already